sync/src/main/java/org/bf2/sync/ManagedKafkaSync.java

```diff
@@ -333,10 +344,35 @@ void create(ManagedKafka remote) {
            if (e.getStatus().getCode() != HttpURLConnection.HTTP_CONFLICT) {
                throw e;
            }
-           log.infof("ManagedKafka %s already exists", Cache.metaNamespaceKeyFunc(remote));
+           // it's possible, but unlikely, that we'll see two reconciliations close enough together
+           // such that we'll naturally see a conflict - that is the second completes before the cache
+           // is updated.  It's more likely that the cache state is not up-to-date, so we'll just
+           // let an exception happen in that case
+           isConsistent(remote);
        }
     }
 
+    /**
+     * Check if the cache state is consistent with the expectation that the entry does not exist in cache
+     * @return true if consistent
+     */
+    private boolean isConsistent(ManagedKafka remote) {
+        ManagedKafka existing = kubeClient.resources(ManagedKafka.class)
+                .inNamespace(remote.getMetadata().getNamespace())
+                .withName(remote.getMetadata().getName()).get();
+        if (existing != null) {
+            checkCreationTimestamp(existing, Duration.ofMinutes(1));
+            throw new RuntimeException(String.format("ManagedKafka %s exists, but not in the informer cache.  The sync may not be in a healthy state", Cache.metaNamespaceKeyFunc(remote)));
```

Comment:
Did you mean to use `InconsistentStateException` here?

Reply:
Sorry, that was an incomplete thought.  I was going to catch that exception for a create conflict, but it really is such a rare occurrence that we might as well let it turn into a full exception - if it is occurring with some regularity it would mean that the async reconcile tasks are being delayed then executed really close together.

Removed now.